File: DahlaFlowerShop/test/keyword-driven/website-keyword-driven.test.js
const fs = require('fs');
const path = require('path');
const { runKeywordDrivenTests } = require('./keyword-driver');

const frontendBaseUrl = process.env.FRONTEND_BASE_URL || 'http://127.0.0.1:5500';
const apiBaseUrl = process.env.API_BASE_URL || 'https://localhost:7114';
const reportsDir = path.join(__dirname, '..', 'reports');
const reportPath = path.join(reportsDir, 'keyword-driven-website-report.json');
const htmlReportPath = path.join(reportsDir, 'keyword-driven-website-report.html');
const testDataPath = path.join(__dirname, 'website-keywords-framework.xlsx');

async function run() {
  fs.mkdirSync(reportsDir, { recursive: true });
  await runKeywordDrivenTests({
    frontendBaseUrl,
    apiBaseUrl,
    reportsDir,
    reportPath,
    htmlReportPath,
    testDataPath
  });
}

run().catch((error) => {
  console.error(error.stack || error.message || error);
  process.exit(1);
});

File: DahlaFlowerShop/test/keyword-driven/website_keyword_driven_test.py
from __future__ import annotations

import os
import sys
from pathlib import Path

from keyword_driver import run_keyword_driven_tests


BASE_DIR = Path(__file__).resolve().parent
TEST_DIR = BASE_DIR.parent
REPORTS_DIR = TEST_DIR / "reports"


def main() -> int:
    summary = run_keyword_driven_tests(
        {
            "frontendBaseUrl": os.getenv("FRONTEND_BASE_URL", "http://127.0.0.1:5500"),
            "apiBaseUrl": os.getenv("API_BASE_URL", "https://localhost:7114"),
            "reportsDir": REPORTS_DIR,
            "reportPath": REPORTS_DIR / "keyword-driven-website-report.json",
            "htmlReportPath": REPORTS_DIR / "keyword-driven-website-report.html",
            "testDataPath": BASE_DIR / "website-keywords-framework.xlsx",
        }
    )
    return 1 if summary["failed"] else 0


if __name__ == "__main__":
    raise SystemExit(main())
